engine/tests/provider/sql_transpiler_test.cpp

```diff
@@ -0,0 +1,129 @@
+        auto ret = ral::io::sql_tools::get_default_operators();
+        for (auto& [op, op_info] : ret) {
+            op_info.parentheses_wrap = false;
+        }
+        ral::io::sql_tools::operator_info oi;
+        oi.parentheses_wrap = false;
+        ret[operator_type::BLZ_LOGICAL_AND] = oi;
+        ret[operator_type::BLZ_LOGICAL_OR] = oi;
+        ret[operator_type::BLZ_ADD] = oi;
+        ret[operator_type::BLZ_SUB] = oi;
+        ret[operator_type::BLZ_MUL] = oi;
+        ret[operator_type::BLZ_DIV] = oi;
+        ret[operator_type::BLZ_LESS] = oi;
+        ret[operator_type::BLZ_LESS_EQUAL] = oi;
+        ret[operator_type::BLZ_GREATER] = oi;
+        ret[operator_type::BLZ_GREATER_EQUAL] = oi;
+        return ret;
+    };
+
+    std::vector<std::string> lineitem_cols = {
+      "l_orderkey", "l_partkey", "l_suppkey", "l_linenumber", "l_quantity",
+      "l_extendedprice", "l_discount", "l_tax", "l_returnflag", "l_linestatus",
+      "l_shipdate", "l_commitdate", "l_receiptdate", "l_shipinstruct",
+      "l_shipmode", "l_comment"};
+    auto all_lineitem_col_idxs = get_all_lineitem_col_idxs(lineitem_cols.size());
+    auto ops_without_pars = get_ops_without_pars();
+
+    auto p = GetParam();
+    this->transformer_type = p.transformer_type;
+    this->queryPart = p.queryPart;
+    this->column_names = p.column_names.empty()? std::move(lineitem_cols) : p.column_names;
+    this->column_indices = p.column_indices.empty()? std::move(all_lineitem_col_idxs) : p.column_indices;
+    this->operators = p.operators.empty()? std::move(ops_without_pars) : p.operators;
+    this->expected = p.expected;
+  }
+
+	void TearDown() override {}
+
+protected:
+  DataType transformer_type;
+  std::string queryPart;
+  std::vector<std::string> column_names;
+  std::vector<int> column_indices;
+  std::map<operator_type, sql_tools::operator_info> operators;
+  std::string expected;
+};
+
+const std::vector<SQLTranspilerTestParam> default_check_entries = {
+  SQLTranspilerTestParam(
+    "BindableTableScan(table=[[main, lineitem]], filters=[[<(+($5, $6), 100)]])",
+    "l_extendedprice + l_discount < 100"
+  ),
+  SQLTranspilerTestParam(
+    "BindableTableScan(table=[[main, lineitem]], filters=[[<(+($5, $6), 100)]])",
+    "((l_extendedprice + l_discount) < 100)",
+    sql_tools::get_default_operators()
+  ),
+  SQLTranspilerTestParam(
+    "BindableTableScan(table=[[main, lineitem_MYSQL]], filters=[[AND(>=($10, 1995-01-01), <=($11, 1996-12-31))]], projects=[[0, 2, 5, 6, 10]], aliases=[[l_orderkey, l_suppkey, l_extendedprice, l_discount, l_shipdate]])",
+    "((l_shipdate >= \"1995-01-01\") AND (l_commitdate <= \"1996-12-31\"))",
+    sql_tools::get_default_operators()
+  ),
+  SQLTranspilerTestParam(
+    "BindableTableScan(table=[[main, lineitem_MYSQL]], filters=[[AND(<($6, $5), IS NOT NULL($0))]], projects=[[0, 11, 12]], aliases=[[l_orderkey, l_commitdate, l_receiptdate]])",
```

Comment:
Same issue here. How is this working? Is it because the transpiler receives a set of column names and it does not know about that was in the project clause? And this example is simply setup for the unit test, and not necessarily "real"

Reply:
If the relational algebra is not "Real"  then i think we should make it real, and have the test could take what is in the project, to then properly obtain what should be set in the column names and column indexes and we can have proper relational algebra in the test. This would make it look more real and also if someone looks at the unit test to better understand the transpiler, then they would know that the column names and indexes it needs to use are not those of the table schema as it is, but based off of the project

Reply:
Yeah, the predicate transpiler only reads the filter part, that's why is working fine, so I think for those unit test we can just change the projection values (i changed the filter variables and forgot to update the projection ones)
cc @gcca

Reply:
@williamBlazing I've updated the projects and aliases fields to consider the column indices in the following commit
https://github.com/BlazingDB/blazingsql/pull/1472/commits/3f51d11852059fbcc9f8cedf5c904a689dea027b